### ISSUE TITLE
set_analog_period changes PWM period for all PWM pins. 

### ISSUE THREAD

**Issue by azamat-bagatov** [bbcmicrobit/micropython#644]
I'm making a synthesizer project, and for it I need to have several different PWM signlas. 
The problem is when you call set_analog_period() on one of the pins it changes it for all PWM pins. Here's an example of the code, that I used to track this:
`pin0.set_analog_period(2)
pin0.write_analog(250)
pin1.set_analog_period(3)
pin1.write_analog(250)
val = 2
while True:
    val = 5-val
    pin1.set_analog_period(val)
    pin1.write_analog(250)
    sleep(2000)`

The output on pin0 and pin1 in this case is identical, although I only call set_analog_period() on pin1. 

If that's not the issue, and you can only change global period, I think this should be highlighted in the documents, as it's far from obvious since it's a pin's method and not microbit's. 
In this case, is there a way to have several pwm signals with different periods? 

Thanks!

**Comment by dpgeorge** [comment]
> and you can only change global period

That's correct.  The PWM driver is a software driver and uses the same timer for all channels.  So all channels must have the same frequency/period (but different duty).  There is no way to change this without a lot of work rewriting the PWM driver.

> I think this should be highlighted in the documents, as it's far from obvious since it's a pin's method and not microbit's.

Yes it would be good to add such a note to the docs.
